openzwave-sys/openzwave-c/manager.cc

```diff
@@ -85,108 +85,133 @@ void manager_set_poll_interval(Manager * manager, int32 interval, bool between_p
+  ValueID vid(cvid->home_id, cvid->id);
+  return stringCreator(manager->GetValueUnits(vid).c_str());
 }
 
-void manager_set_value_units(Manager * manager, const ValueID * id, char const * str) {
+void manager_set_value_units(Manager * manager, const cValueID * cvid, char const * str) {
+  ValueID vid(cvid->home_id, cvid->id);
   const std::string string(str);
-  manager->SetValueUnits(*id, string);
+  manager->SetValueUnits(vid, string);
 }
 
-char * manager_get_value_help(Manager * manager, const ValueID * id, const RustStringCreator stringCreator) {
-  return stringCreator(manager->GetValueHelp(*id).c_str());
+char * manager_get_value_help(Manager * manager, const cValueID * cvid, const RustStringCreator stringCreator) {
+  ValueID vid(cvid->home_id, cvid->id);
+  return stringCreator(manager->GetValueHelp(vid).c_str());
 }
 
-void manager_set_value_help(Manager * manager, const ValueID * id, char const * str) {
+void manager_set_value_help(Manager * manager, const cValueID * cvid, char const * str) {
+  ValueID vid(cvid->home_id, cvid->id);
   const std::string string(str);
-  manager->SetValueHelp(*id, string);
+  manager->SetValueHelp(vid, string);
 }
 
-int32 manager_get_value_min(Manager * manager, const ValueID * id) {
-  return manager->GetValueMin(*id);
+int32 manager_get_value_min(Manager * manager, const cValueID * cvid) {
+  ValueID vid(cvid->home_id, cvid->id);
+  return manager->GetValueMin(vid);
 }
 
-int32 manager_get_value_max(Manager * manager, const ValueID * id) {
-  return manager->GetValueMax(*id);
+int32 manager_get_value_max(Manager * manager, const cValueID * cvid) {
+  ValueID vid(cvid->home_id, cvid->id);
+  return manager->GetValueMax(vid);
 }
 
-bool manager_is_value_read_only(Manager * manager, const ValueID * id) {
-  return manager->IsValueReadOnly(*id);
+bool manager_is_value_read_only(Manager * manager, const cValueID * cvid) {
+  ValueID vid(cvid->home_id, cvid->id);
+  return manager->IsValueReadOnly(vid);
 }
 
-bool manager_is_value_write_only(Manager * manager, const ValueID * id) {
-  return manager->IsValueWriteOnly(*id);
+bool manager_is_value_write_only(Manager * manager, const cValueID * cvid) {
+  ValueID vid(cvid->home_id, cvid->id);
+  return manager->IsValueWriteOnly(vid);
 }
 
-bool manager_is_value_set(Manager * manager, const ValueID * id) {
-  return manager->IsValueSet(*id);
+bool manager_is_value_set(Manager * manager, const cValueID * cvid) {
+  ValueID vid(cvid->home_id, cvid->id);
+  return manager->IsValueSet(vid);
 }
 
-bool manager_is_value_polled(Manager * manager, const ValueID * id) {
-  return manager->IsValuePolled(*id);
+bool manager_is_value_polled(Manager * manager, const cValueID * cvid) {
+  ValueID vid(cvid->home_id, cvid->id);
+  return manager->IsValuePolled(vid);
 }
 
 GET_VALUE_FUNC(as_bool, bool* value) {
-  return manager->GetValueAsBool(*id, value);
+  ValueID vid(cvid->home_id, cvid->id);
```

Comment:
nit: Maybe you could have done a GET_VALUE_FUNC_IMPL macro to help removing some repetition here. But as you wish.